### PR TITLE
control parameters in Thunder

### PR DESCRIPTION
**Pull request by yammineramy**
updates to our version of ompl needed to allow for calls to change thunder parameters from dig.

### CODE REVIEW

File: src/ompl/tools/thunder/src/Thunder.cpp [hunk 1]
```diff
@@ -515,3 +515,11 @@ bool ompl::tools::Thunder::doPostProcessing()
 
     return true;
 }
+
+void ompl::tools::Thunder::setStretchFactor(double stretch_factor){stretch_factor_ = stretch_factor;};
+double ompl::tools::Thunder::getStretchFactor(){return stretch_factor_;};
+void ompl::tools::Thunder::setSparseDelta(double SparseD){SparseD_ = SparseD;};
+double ompl::tools::Thunder::getSparseDelta(){return SparseD_;};
+void ompl::tools::Thunder::setDenseDelta(double DenseD){DenseD_ = DenseD;};
+double ompl::tools::Thunder::getDenseDelta(){return DenseD_;};
```
Comment (duanyutong) [inline src/ompl/tools/thunder/src/Thunder.cpp:524]: move to header

File: src/ompl/tools/thunder/Thunder.h [hunk 2]
```diff
@@ -97,6 +97,9 @@ namespace ompl
         private:
             /** \brief Shared constructor functions */
             void initialize();
+            double stretch_factor_ {1.2};
+            double DenseD_ {0.001};
+            double SparseD_ {0.1};
```
Comment (duanyutong) [inline src/ompl/tools/thunder/Thunder.h:102]: where are these used?

File: src/ompl/tools/thunder/src/Thunder.cpp (hunk 1 above)
Comment (yammineramy) [reply src/ompl/tools/thunder/src/Thunder.cpp]: Okay, done in new version.

File: src/ompl/tools/thunder/src/Thunder.cpp [hunk 3]
```diff
@@ -81,18 +80,23 @@ void ompl::tools::Thunder::setup()
...
+                    OMPL_INFORM("Planner Allocator specified");
+                    planner = pa_(si_);
+                }
+                else
+                {
+                    planner = std::make_shared<ompl::geometric::RRTConnect>(si_);
+                }
+                planner->setProblemDefinition(pdef_);
+                if (!planner->isSetup())
+                    planner->setup();
```
Comment (duanyutong) [inline src/ompl/tools/thunder/src/Thunder.cpp:98]: use `.clang-format` to format your code.

File: src/ompl/tools/thunder/src/Thunder.cpp [hunk 4]
```diff
@@ -131,10 +136,10 @@ void ompl::tools::Thunder::setup()
...
             experienceDB_->getSPARSdb()->setup();
 
-            experienceDB_->getSPARSdb()->setStretchFactor(1.6); 
-            experienceDB_->getSPARSdb()->setSparseDeltaFraction(
-                0.1);  // was 0.05 // vertex visibility range  = maximum_extent * this_fraction
-            experienceDB_->getSPARSdb()->setDenseDeltaFraction(0.001);
+            experienceDB_->getSPARSdb()->setStretchFactor(getStretchFactor());
+            experienceDB_->getSPARSdb()->setSparseDeltaFraction(getSparseDelta());  // was 0.05 // vertex visibility range  =
+                                                                       // maximum_extent * this_fraction
+            experienceDB_->getSPARSdb()->setDenseDeltaFraction(getDenseDelta());
```
Comment (sadraddini) [inline src/ompl/tools/thunder/src/Thunder.cpp:142]: Are these inside the class methods? You don't need to call "get" functions to access private variables. Simply have 
`experienceDB_->getSPARSdb()->setStretchFactor(stretch_factor_)`

File: src/ompl/tools/thunder/Thunder.h (hunk 2 above)
Comment (sadraddini) [reply src/ompl/tools/thunder/Thunder.h]: Also not a terrible idea to add some comments explaining what these are.

**Comment by sadraddini** [comment]
Are we all good on this?

**Comment by sadraddini** [comment]
@duanyutong Is there anything you want to change before merging?

File: src/ompl/tools/thunder/Thunder.h [hunk 5]
```diff
@@ -33,7 +33,7 @@
  *********************************************************************/
 
 /* Author: Dave Coleman
-*/
+ */
```
Comment (duanyutong) [inline src/ompl/tools/thunder/Thunder.h:36]: remove any whitespace changes w.r.t. upstream

File: src/ompl/tools/thunder/Thunder.h [hunk 6]
```diff
@@ -136,9 +139,10 @@ namespace ompl
             /** \brief Run the planner until \e ptc becomes true (at most) */
             base::PlannerStatus solve(const base::PlannerTerminationCondition &ptc) override;
 
-            base::PlannerStatus solve(const base::PlannerTerminationCondition &ptc, bool hybridize) {
-              hybridize_ = hybridize;
-              return solve(ptc);
+            base::PlannerStatus solve(const base::PlannerTerminationCondition &ptc, bool hybridize)
+            {
+                hybridize_ = hybridize;
+                return solve(ptc);
```
Comment (duanyutong) [inline src/ompl/tools/thunder/Thunder.h:145]: revert formatting changes. please use `.clang-format`

File: src/ompl/tools/thunder/Thunder.h [hunk 7]
```diff
@@ -183,16 +187,42 @@ namespace ompl
             /** \brief Allow accumlated experiences to be processed */
             bool doPostProcessing() override;
 
-            void addSolutionToQueue(const std::shared_ptr<ompl::geometric::PathGeometric>& path_ptr) {
-              queuedSolutionPaths_.push_back(*path_ptr); 
+            void addSolutionToQueue(const std::shared_ptr<ompl::geometric::PathGeometric> &path_ptr)
+            {
+                queuedSolutionPaths_.push_back(*path_ptr);
             }
```
Comment (duanyutong) [inline src/ompl/tools/thunder/Thunder.h:193]: revert formatting changes. please use .clang-format

**Review by duanyutong — COMMENTED** [review]
were the formatting changes made by this repo's `.clang-format`? we should commit formatting changes ONLY IF that is the case. even then, clang-formatting the codebase should be a no-op PR, separate from actual code changes.

File: src/ompl/tools/thunder/Thunder.h (hunk 7 above)
Comment (yammineramy) [reply src/ompl/tools/thunder/Thunder.h:193]: I use the command clang-format -i Thunder.cc
Isn't this what we want?

Comment (duanyutong) [reply src/ompl/tools/thunder/Thunder.h:193]: no, you have to use the `.clang-format` config

**Review by duanyutong — APPROVED** [review]
/lgtm

**Review by sadraddini — APPROVED** [review]

**Pull request merged**